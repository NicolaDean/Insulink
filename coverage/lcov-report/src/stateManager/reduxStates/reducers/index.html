
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/stateManager/reduxStates/reducers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../index.html">All files</a> src/stateManager/reduxStates/reducers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.21% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>144/171</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">54.34% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>25/46</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">73.91% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>17/23</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.61% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>143/169</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="macroTracker.js"><a href="macroTracker.js.html">macroTracker.js</a></td>
	<td data-value="88.61" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 88%"></div><div class="cover-empty" style="width: 12%"></div></div>
	</td>
	<td data-value="88.61" class="pct high">88.61%</td>
	<td data-value="123" class="abs high">109/123</td>
	<td data-value="52.94" class="pct medium">52.94%</td>
	<td data-value="34" class="abs medium">18/34</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="15" class="abs high">12/15</td>
	<td data-value="89.34" class="pct high">89.34%</td>
	<td data-value="122" class="abs high">109/122</td>
	</tr>

<tr>
	<td class="file medium" data-value="userReducer.js"><a href="userReducer.js.html">userReducer.js</a></td>
	<td data-value="72.34" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 72%"></div><div class="cover-empty" style="width: 28%"></div></div>
	</td>
	<td data-value="72.34" class="pct medium">72.34%</td>
	<td data-value="47" class="abs medium">34/47</td>
	<td data-value="58.33" class="pct medium">58.33%</td>
	<td data-value="12" class="abs medium">7/12</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	<td data-value="71.73" class="pct medium">71.73%</td>
	<td data-value="46" class="abs medium">33/46</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Thu May 26 2022 11:26:03 GMT+0200 (Central European Summer Time)
            </div>
        <script src="../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../sorter.js"></script>
        <script src="../../../../block-navigation.js"></script>
    </body>
</html>
    