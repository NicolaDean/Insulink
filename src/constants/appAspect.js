import { Dimensions  } from 'react-native';
// primary in hex: #70CAE6
export const themeBlue = {
    white:"#fff",
    black:"black",
    primary:"rgb(112,202,230)",
    secondary:"",

};

export const colors = themeBlue;


const dimension = Dimensions.get('window');

export const dim = {
    height: dimension.height,
    width : dimension.width,
}