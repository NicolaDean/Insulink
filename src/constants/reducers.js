export const foodMethods = {
    addFood:'ADD_FOOD',
    removeFood:'REMOVE_FOOD',
    selectMeal:'SELECT_MEAL',
}

export const userMethods = {
    registerUser: 'REGISTER_USER',
    deleteUser: 'DELETE_USER',
    logout: 'LOGOUT_USER',
    login: 'LOGIN_USER',
    updateUser: 'UPDATE_USER',
    addGlicemy: 'ADD_GLICEMY',
}