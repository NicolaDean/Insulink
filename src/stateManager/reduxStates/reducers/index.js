import macroReducer from "./macroTracker";
import { combineReducers } from "redux";

const appReducers = combineReducers({
    macroTracker : macroReducer,
});

export default appReducers;