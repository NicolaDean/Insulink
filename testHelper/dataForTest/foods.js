
export const dummyAppleDeletable = {"id":{"food_name":"apple pie","serving_unit":"piece (1/8 of 9\" dia)","tag_name":"Apple pie","serving_qty":1,"common_type":null,"tag_id":"15060","photo":{"thumb":"https://nix-tag-images.s3.amazonaws.com/1291_thumb.jpg"},"locale":"en_US"},"name":"apple pie","image":"https://nix-tag-images.s3.amazonaws.com/1291_highres.jpg","cal":237,"carb":34,"fat":11,"prot":1.9,"quantity":1,"unit":"g","identifier":1}

export const dummyApple = {"food_name":"apple pie","serving_unit":"piece (1/8 of 9\" dia)","tag_name":"Apple pie","serving_qty":1,"common_type":null,"tag_id":"15060","photo":{"thumb":"https://nix-tag-images.s3.amazonaws.com/1291_thumb.jpg"},"locale":"en_US"}
