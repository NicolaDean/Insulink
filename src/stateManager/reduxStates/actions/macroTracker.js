import { foodMethods } from "../../../constants/reducers"

export const addFood = (food) =>{
    return{
        type: foodMethods.addFood,
        payload: {food:food}
    }
} 

export const removeFood = (food,mealType) =>{
    return{
        type: foodMethods.removeFood,
        payload: {food:food,mealType:mealType}
    }
} 

export const selectMealType = (mealType) =>{
    return{
        type: foodMethods.selectMeal,
        payload: {mealType:mealType}
    }
}