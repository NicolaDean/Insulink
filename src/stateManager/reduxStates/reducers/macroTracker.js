
import { macroConstants } from "../../../constants/states";
import { foodMethods } from "../../../constants/reducers"
import { FirebaseQuery } from "../../../utils/firebaseQuery";

const macro = macroConstants;

//TODO Each diary will have a Date, if Date < Today it will be saved localy/firabase and resetted for new day
export const initialDiaryState = {
    currentMeal:"breakfast",
    currentDate:FirebaseQuery.glicemyDateFormatter(new Date()),
    totMacro:{cal:0,carb:0,fat:0,prot:0},
    id:1,
    activity_id:1,
    meals:{
        breakfast:{foods:[],macro:{cal:0,carb:0,fat:0,prot:0}},
        lunch:{foods:[],macro:{cal:0,carb:0,fat:0,prot:0}},
        dinner:{foods:[],macro:{cal:0,carb:0,fat:0,prot:0}},
        snack:{foods:[],macro:{cal:0,carb:0,fat:0,prot:0}}
    },
    history:{
        empty:true,
        totMacro:{cal:0,carb:0,fat:0,prot:0},
        meals:{
            breakfast:{foods:[],macro:{cal:0,carb:0,fat:0,prot:0}},
            lunch:{foods:[],macro:{cal:0,carb:0,fat:0,prot:0}},
            dinner:{foods:[],macro:{cal:0,carb:0,fat:0,prot:0}},
            snack:{foods:[],macro:{cal:0,carb:0,fat:0,prot:0}}
        },
        activities:{
            breakfast:{sports:[],totCal:0},
            lunch:{sports:[],totCal:0},
            dinner:{sports:[],totCal:0},
            snack:{sports:[],totCal:0}
        }
    },
    totCalBurned:0,
    activities:{
        breakfast:{sports:[],totCal:0},
        lunch:{sports:[],totCal:0},
        dinner:{sports:[],totCal:0},
        snack:{sports:[],totCal:0}
    }
}

/**
 * Select current meal to assign added food to
 * @param {*} state 
 * @param {*} data 
 * @returns 
 */
const selectMeal = (state,data)=>{
    const newState = {
        ...state,
        currentMeal: data.mealType}

    console.log("Current Meal:" + state.currentMeal);
    return newState;
}

/**
 * Merge 2 macro into a single one by summing its values
 * @param {*} base first macro
 * @param {*} toSum second macro
 * @returns merged macros
 */
const sumMacro = (base,toSum) =>{

    let result = {};

    result[macro.cal] = base[macro.cal] + toSum[macro.cal];
    result[macro.fat] = base[macro.fat] + toSum[macro.fat];
    result[macro.carb] = base[macro.carb] + toSum[macro.carb];
    result[macro.prot] = base[macro.prot] + toSum[macro.prot];

    console.log("ADDING THIS MACRO: " + base[macro.cal] +  " + " + toSum[macro.cal] +"=" +  result[macro.cal]);
   
    return result;
}
/**
 * Merge 2 macro into a single one by subbing its values
 * @param {*} base first macro
 * @param {*} toSum second macro
 * @returns merged macros
 */
const subMacro = (base,toRemove) =>{

    let result = {};

    
    result[macro.cal] = base[macro.cal] - toRemove[macro.cal];
    result[macro.fat] = base[macro.fat] - toRemove[macro.fat];
    result[macro.carb] = base[macro.carb] - toRemove[macro.carb];
    result[macro.prot] = base[macro.prot] - toRemove[macro.prot];

    console.log("REMOVING THIS MACRO: " + base[macro.cal] +  " - " + toRemove[macro.cal] +"=" +  result[macro.cal]);
    return result;
}

/**
 * add a food to the global state representing meals of today
 * @param {*} state today meals
 * @param {*} data food to add
 * @returns new state with the new food added to the current selected meal
 */
const addFood = (state,data)=>{
    
    const food = data.food;
    //COPY STATE
    const newstate ={...state};
    food.identifier = newstate.id;
    //INCREMENT ID
    newstate.id = newstate.id + 1;
    //PUSH FOOD

    newstate.meals[state.currentMeal].foods.push(data.food);
    //CALCULATE TOT MACRO
    let m = state.meals[state.currentMeal].macro;
    newstate.totMacro  = sumMacro(newstate.totMacro,data.food);
    newstate.meals[state.currentMeal].macro = sumMacro(m,data.food);

    return newstate;
}

const addActivity = (state,data) =>{
    const activity = data.activity;

    //COPY STATE
    const newstate ={...state};
    activity.identifier = newstate.activity_id;
    newstate.activity_id = newstate.activity_id+1;
    //
    newstate.activities[newstate.currentMeal].sports.push(activity);
    newstate.activities[newstate.currentMeal].totCal += activity.calories;
    newstate.totCalBurned += activity.calories;

    return newstate;
}



const findMatchingIdMeal = (array,identifier,isSport = false) =>{

    console.log("Identifier: " + identifier)
    //REMOVING FOOD FROM LIST:
    let found = false;
    let m_found = false;

    let meal_types = Object.keys(array);

    //FOR EACH MEAL
    meal_types.forEach((key,index) =>{
        if(found != false){ return;}
            let i = 0;
            let tmp = [];
            if(isSport) {
                tmp =array[key].sports;
            }else{
                tmp =array[key].foods;
            }
            
    
            //FOR EACH FOOD
            tmp.forEach(x =>{
                //CHECK IDENTIFIER TO REMOVE
                console.log("(" + x.identifier + " , " + identifier+ ")");
            
                if(x.identifier === identifier)
                {
                    //IF FOUNDED MARK INDEX AND MEAL FROM WICH REMOVE
                    found = i;
                    m_found = key;
                }
                i++;
            });        
        });

        return {index:found,meal:m_found};
}
/**
 * remove a food to the global state representing meals of today
 * @param {*} state today meals
 * @param {*} data food to add
 * @returns new state with the selected food removed to the current selected meal
 */
const removeFood = (state,data)=>{

    const food = data.food;
    //COPY STATE
    const newstate ={...state};
   
    console.log("FOOOD: " + JSON.stringify(food));

    //FIND MATCHING FOOD IN STATE
    let matching = findMatchingIdMeal(newstate.meals,food.identifier,false);

    //REMOVE FROM MEAL THE FOOD AT THE SPECIFIC INDEX MARKED
    if((matching.index-1 === -1) || matching.index != false){
        console.log("REMOVING: " + JSON.stringify(food));

        //REMOVING MACRO
        let m = state.meals[matching.meal].macro;
        newstate.meals[matching.meal].macro = subMacro(m,food);
        newstate.totMacro  = subMacro(newstate.totMacro,food);

        //REMOVING FOOD FROM LIST
        newstate.meals[matching.meal].foods.splice(matching.index,1);
        
    }
    

    return newstate;
}

const editFood = (state,data) =>{
    const food = data.food;
    //COPY STATE
    const newstate ={...state};
   
    //FIND MATCHING FOOD IN STATE
    let matching = findMatchingIdMeal(newstate.meals,food.identifier,false);


    //REMOVE FROM MEAL THE FOOD AT THE SPECIFIC INDEX MARKED
    if((matching.index-1 === -1) || matching.index != false){
        console.log("FOUND IN (" +matching.meal + ","+matching.index+")" );
        //PROBLEM IS IN IDENTIFIER!!
        let m = newstate.meals[matching.meal].macro;
        console.log(JSON.stringify(newstate));
        let oldFood = newstate.meals[matching.meal].foods[matching.index];

        //REMOVE FOOD FROM MACRO
        //(BISOGNA PASSARE IL CIBO VERO E PROPRIO MO VIENE PASSATO L'ID)
        //Remove old version Macro
        newstate.meals[matching.meal].macro = subMacro(m,oldFood);
        newstate.totMacro  = subMacro(newstate.totMacro,oldFood);

        //Add new Version Macro
        newstate.meals[matching.meal].macro = sumMacro(newstate.meals[matching.meal].macro,food);
        newstate.totMacro  = sumMacro(newstate.totMacro,food);

        newstate.meals[matching.meal].foods[matching.index] = food;
    }
    

    return newstate;
}

const removeActivity = (state,data) =>{
    //COPY STATE
    const newstate ={...state};
    const activity = data.activity;


    //FIND MATCHING ACTIVITY IN STATE
    let matching = findMatchingIdMeal(newstate.activities,activity.identifier,true);


    if((matching.index-1 === -1) || matching.index != false){
        //TODO REMOVE CALORIES
        newstate.activities[matching.meal].totCal -= activity.calories;
        newstate.totCalBurned -= activity.calories;
        //REMOVING FOOD FROM LIST
        newstate.activities[matching.meal].sports.splice(matching.index,1);
    }
    //TODO REMOVE ACTIVITY
    return newstate;
}
/**
 * Load Meals data from memory
 * @param {*} state 
 * @param {*} payload 
 * @returns 
 */
const loadMeals = (state,payload) =>{
    return payload.diary;
}

/**
 * reset the diary for a new day
 * @param {*} state 
 * @param {*} payload 
 * @returns 
 */
const resetDiary = (state,payload) =>{

    const newState = {...initialDiaryState};

    console.log("UFF: " + JSON.stringify(newState))

    newState.currentDate = payload.currentDate;

    return newState;
}

const loadHistory = (state,payload) => {
    const newState = {...state};

    newState.currentDate = payload.date;
    if(payload.date != FirebaseQuery.glicemyDateFormatter()){
        if(payload.history!=null){
            newState.history = payload.history;
        }
    }
    
    return newState;
}

//TODO PUT THE NAME OF ACTIONS INTO COSNTANT FILE
const macroReducer = (state = initialDiaryState, action) => {
    switch(action.type){
        case foodMethods.addFood:
            return addFood(state,action.payload);
        case foodMethods.removeFood:
            return removeFood(state,action.payload);
        case foodMethods.editFood:
            return editFood(state,action.payload);
        case foodMethods.selectMeal:
            return selectMeal(state,action.payload);
        case foodMethods.loadFoodDiary:
            return loadMeals(state,action.payload);
        case foodMethods.addActivity:
            return addActivity(state,action.payload);
        case foodMethods.resetDiary:
            return resetDiary(state,action.payload);
        case foodMethods.removeActivity:
            return removeActivity(state,action.payload);
        case foodMethods.loadHistory:
            return loadHistory(state,action.payload);
        default:
            return state;
        }
}

export default macroReducer;