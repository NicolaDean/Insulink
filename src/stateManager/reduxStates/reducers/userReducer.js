import { userMethods } from "../../../constants/reducers"
import {  FirebaseQuery } from "../../../utils/firebaseQuery";
import { loginStatus } from "../../../constants/states";

export const initialState = {
    status:loginStatus.unlogged,
    userId:"nullId",
    mustCompleteReg:false,
    userData:{
        email:"",
        password:"",
        name:"dummy",
        surname:"",
        weight:80,
        height:180,
        birthday:{seconds:0,nanoseconds:0},
        isf:0,
        choratio:0,
        glicemy:[],
        activitys:undefined,
        maxCarb:200,
        maxFat:100,
        maxProt:40,
    }
}

const login = (state,payload) =>{
    const usrData = payload.usrData;

    const newState = {...state};

    console.log("THIS USER HAS : " +payload.userId )
    newState.userId = payload.userId;
    newState.status = loginStatus.logged;
    newState.userData = usrData;
    return newState;
}

const del = (state,payload) =>{
    
}

const logout = (state,payload) =>{
    const newState = {...initialState};

    return {...initialState};
}


const register = (state,payload) =>{
    const newstate ={...state};
    newstate.userId = payload.userId;
    newstate.userData = payload.user;
    newstate.status = loginStatus.logged;
    newstate.mustCompleteReg = false;
    
    return newstate;
}

const addGlicemy = (state,payload) =>{
    const newstate ={...state};

    const id = FirebaseQuery.glicemyDateFormatter();

    const g = {...payload.glicemy};//;

    if(newstate.userData.glicemy[id] == undefined) newstate.userData.glicemy[id] = [];

    newstate.userData.glicemy[id].push(g);
    //console.log(newstate.userData.glicemy[id]);
    return newstate;
}

const update = (state,payload) =>{
    const newstate = {...state};

    newstate.userData = payload.user;

    return newstate;
}

const setId = (state,payload) =>{
    const newstate = {...state};

    newstate.userId = payload.uid;
    newstate.mustCompleteReg = true;

    return newstate;
}

const addActivity = (state,payload) =>{
    const newstate = {...state};

    const cal = payload.activity.calories;
    const today = FirebaseQuery.glicemyDateFormatter(new Date(),true);

    let pos = -1;
    let i = 0;
    let x = [];

    if(newstate.userData.activitys == undefined) newstate.userData.activitys = [];
    console.log("ACTIV: " + JSON.stringify(newstate.userData.activitys));
    newstate.userData.activitys.forEach(act=>{
        if(act.date == today){
            pos = i;
        }
        i++;
    });

    if(pos != -1){
        newstate.userData.activitys[pos].count += cal;
    }else{
        newstate.userData.activitys.push({date: today, count: cal});
    }

    console.log("NEW STATE: " + JSON.stringify(newstate.userData.activitys));

    return newstate;
}

const removeActivity = (state,payload) =>{
    const newstate = {...state};
    const today = FirebaseQuery.glicemyDateFormatter(new Date(),true);
    const cal = payload.activity.calories;
    let pos = -1;
    let i = 0;
    
    if(newstate.userData.activitys == undefined) newstate.userData.activitys = [];
    console.log("ACTIV: " + JSON.stringify(newstate.userData.activitys));
    newstate.userData.activitys.forEach(act=>{
        if(act.date == today){
            pos = i;
        }
        i++;
    });

    if(pos != -1){
        newstate.userData.activitys[pos].count -= cal;
    }
    return newstate;
}

const userReducer = (state = initialState, action) => {
    switch(action.type){
        case userMethods.login:
            return login(state,action.payload);
        case userMethods.setId:
            return setId(state,action.payload);
        case userMethods.deleteUser:
            return del(state,action.payload);
        case userMethods.registerUser:
            return register(state,action.payload);
        case userMethods.updateUser:
            return update(state,action.payload);
        case userMethods.addGlicemy:
            return addGlicemy(state,action.payload);
        case userMethods.addActivity:
            return addActivity(state,action.payload);
        case userMethods.removeActivity:
            return removeActivity(state,action.payload);
        case userMethods.logout:
            return logout(state,action.payload);
        default:
            return state;
    }
}

export default userReducer;