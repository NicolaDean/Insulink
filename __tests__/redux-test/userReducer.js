import React from 'react';
import { foodMethods, userMethods } from '../../src/constants/reducers';
import macroTracker, { initialState } from '../../src/stateManager/reduxStates/reducers/macroTracker';
import userReducer from '../../src/stateManager/reduxStates/reducers/userReducer';
import { dummyApple } from '../../testHelper/dataForTest/foods';

const fir = ()=>{
    return {
      doc:null,
      collection: (a) =>{return "ciao"},
    };
}

jest.mock("@react-native-firebase/firestore", () => fir);




/**
 * Those test allow to check if the reducer change correctly the global state after an action
 */
describe("TESTING THE REDUX REDUCER FOR USER:", ()=>{

    beforeAll(()=>{
        Date.now = jest.fn(() => 1482363367071);
    });

    afterAll(()=>{
        Date.now = jest.fn(() => 1482363367071);
    });

    //ALL user Reducer test
    test("Login,Update,Glicemy,Logout",()=>{
        const userData = {name:"test",height:77,glicemy: []}
        const userId = "ABCDE123456";
        const action ={
            type:userMethods.login,
            payload:{
                usrData:userData,
                userId:userId
            }
        }

        let prevState; //State of previous operation
        //TRY LOGIN
        expect(prevState = userReducer(initialState,action)).toMatchSnapshot();
        
        //UPDATE USER
        const newUser = {...userData,name:"Pippo"}
        action.type = userMethods.updateUser;
        action.payload = {
            user:newUser
        }
        expect(prevState = userReducer(prevState,action)).toMatchSnapshot();
        
        //TRY ADD GLICEMY
        const glicemy = {value:43,time:{hours:"11:11",day:"22/1"}}
        action.type = userMethods.addGlicemy;
        action.payload = {
            glicemy:glicemy
        }
        expect(prevState = userReducer(prevState,action)).toMatchSnapshot();
        
        console.log(prevState);
        //LOGOUT
        action.type = userMethods.logout;
        expect(prevState = userReducer(prevState,action)).toMatchSnapshot();
    });


})