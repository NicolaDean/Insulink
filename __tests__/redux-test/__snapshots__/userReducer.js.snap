// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING THE REDUX REDUCER FOR USER: Login,Update,Glicemy,Logout 1`] = `
Object {
  "mustCompleteReg": false,
  "status": "LOGGED",
  "userData": Object {
    "glicemy": Array [],
    "height": 77,
    "name": "test",
  },
  "userId": "ABCDE123456",
}
`;

exports[`TESTING THE REDUX REDUCER FOR USER: Login,Update,Glicemy,Logout 2`] = `
Object {
  "mustCompleteReg": false,
  "status": "LOGGED",
  "userData": Object {
    "glicemy": Array [],
    "height": 77,
    "name": "Pippo",
  },
  "userId": "ABCDE123456",
}
`;

exports[`TESTING THE REDUX REDUCER FOR USER: Login,Update,Glicemy,Logout 3`] = `
Object {
  "mustCompleteReg": false,
  "status": "LOGGED",
  "userData": Object {
    "glicemy": Array [],
    "height": 77,
    "name": "Pippo",
  },
  "userId": "ABCDE123456",
}
`;

exports[`TESTING THE REDUX REDUCER FOR USER: Login,Update,Glicemy,Logout 4`] = `
Object {
  "mustCompleteReg": false,
  "status": "UNLOGGED",
  "userData": Object {
    "birthday": Object {
      "nanoseconds": 0,
      "seconds": 0,
    },
    "choratio": 0,
    "email": "",
    "glicemy": Array [],
    "height": 180,
    "isf": 0,
    "maxCarb": 200,
    "maxFat": 100,
    "maxProt": 40,
    "name": "dummy",
    "password": "",
    "surname": "",
    "weight": 80,
  },
  "userId": "nullId",
}
`;
